src/shape.cpp

```diff
@@ -195,6 +195,14 @@ bool shape::scalar() const
 
 bool shape::standard() const { return impl->m_standard; }
 
+shape shape::normalize_standard() const
+{
+    if(this->standard())
```

Comment:
This function is going to implicitly convert a scalar type to the shape [1] with stride [1]

Reply:
Yes, I think it should be expected. This is used to pass the shapes to miopen tensor descriptors but it doesn't support zero strides.